### PR TITLE
Apply flood workspace algorithm

### LINKED ISSUE #23318: Algorithm to apply a flood correction workspace.
Add an algorithm that applies flood correction using workspace created by `CreateFloodWorkspace` from #23212.

This is a part of #12524

### PR DESCRIPTION
**Pull request by mantid-roman**
To be merged after https://github.com/mantidproject/mantid/pull/23317 and https://github.com/mantidproject/mantid/pull/23337

**Description of work.**
Added algorithm ApplyFloodWorkspace.

**To test:**
Run this script:
```
flood = CreateFloodWorkspace('OFFSPEC00035946.nxs', StartSpectrumIndex=250, EndSpectrumIndex=600,
                             Exclude=[260, 261, 262, 516, 517, 518], OutputWorkspace='flood')
data = Load('OFFSPEC00035946.nxs')
ApplyFloodWorkspace(InputWorkspace='data', FloodWorkspace='flood', OutputWorkspace='out')
```
If plotted the first bin of "out" workspace should have a straight line in the middle of the graph.

Fixes #23318 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: docs/source/algorithms/ApplyFloodWorkspace-v1.rst [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+.. summary::
+
+.. relatedalgorithms::
+
+.. properties::
+
+Description
+-----------
+
+The algorithm applies flood correction to a workspace by dividng it by a flood workspace. The flood workspace can be created with
```
Comment (Mantid-Matthew) [inline docs/source/algorithms/ApplyFloodWorkspace-v1.rst]: missing an i from diving here

**Comment by Mantid-Matthew** [comment]
All looks good to me just one small typo in documentation then should be good to go.

**Review by Matthew-Andrew — APPROVED** [review]
Looks good to me

**Comment by gvardany** [comment]
Just to confirm, is the main purpose of this to be an algorithm is to change the event type? Since otherwise it does nothing but Divide.

**Comment by mantid-roman** [comment]
@gvardany It names the procedure: it applies the flood correction. It does whatever needed to be done. At the moment it is division by a workspace. But it allows the scientists to change their mind in the future and use multiplication, for example. If it turns out rebinning or something else is necessary, it can be included later.

**Comment by gvardany** [comment]
I see, if there are prospects of this to be evolved later, that is justifiable. But in general, I think we should be a bit cautious when adding new algorithms just for naming procedures.

**Pull request merged**